Backpack/Icon/Classes/BPKIcon.m

```diff
@@ -0,0 +1,236 @@
+
+    return _paragraphStyle;
+}
+
++ (nullable NSDictionary<NSString *, NSString *> *)iconMapping {
+    static NSDictionary<NSString *, NSString *> *_iconMapping;
+    static dispatch_once_t dispatchOnceiconMappingToken;
+
+    dispatch_once(&dispatchOnceiconMappingToken, ^{
+        NSBundle *bundle = [self iconBundle];
+        if (bundle == nil) {
+            _iconMapping = nil;
+            return;
+        }
+
+
+        NSURL *url = [bundle URLForResource:@"iconMapping" withExtension:@"json"];
+        NSAssert(url, @"`iconMapping.json` must be present in the bundle.");
+        if (url == nil) {
+            _iconMapping = nil;
+            return;
+        }
+
+        NSData *data = [NSData dataWithContentsOfURL:url];
+        NSAssert1(data, @"`iconMapping.json` at %@ could not be loaded from the bundle", [url description]);
+
+        NSError *error = nil;;
+        NSDictionary *parsedData = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
+        NSAssert(!error, @"Expected `iconMapping.json` to be a valid JSON document, but got error %@", error);
+        NSAssert([parsedData isKindOfClass:[NSDictionary class]], @"Expectd `iconMapping.json` to be a dictionary");
+        if (error || parsedData == nil || ![parsedData isKindOfClass:[NSDictionary class]]) {
+            _iconMapping = nil;
+            return;
+        }
+
+        _iconMapping = parsedData;
+    });
+
+    return _iconMapping;
+}
+
++ (NSBundle *)iconBundle {
+    NSBundle *mainBundle = [NSBundle bundleForClass:[self class]];
+    NSAssert(mainBundle, @"bundleForClass:[BPKIcon class] should return a valid bundle");
+
+    NSURL *iconBundleURL = [[mainBundle resourceURL] URLByAppendingPathComponent:@"Icon.bundle"];
+    NSBundle *iconBundle = [NSBundle bundleWithURL:iconBundleURL];
+    NSAssert(iconBundle, @"The Backpack Icon bundle should exist at %@", iconBundleURL);
+
+    return iconBundle;
+}
+
++ (NSString *)cacheKeyForIconNamed:(NSString *)name withColor:(UIColor *)color size:(BPKIconSize)size {
+    NSString *sizeName;
+
+    switch (size) {
+        case BPKIconSizeSmall:
+            sizeName = @"sm";
+            break;
+        case BPKIconSizeLarge:
+            sizeName = @"lg";
+            break;
+        default:
+            NSAssert(NO, @"Unknown icon size");
+            sizeName = @"unknown";
+    }
+
+    CGFloat const *components = CGColorGetComponents(color.CGColor);
+
+    return [NSString stringWithFormat:@"%@%@%f%f%f%f", name, sizeName, components[0], components[1], components[2], components[3]];
+}
+
++ (CGSize)concreteSizeForIconSize:(BPKIconSize)size {
+    switch (size) {
+        case BPKIconSizeSmall:
+            return CGSizeMake(16, 16);
+        case BPKIconSizeLarge:
+            return CGSizeMake(24, 24);
+        default:
+            NSAssert(NO, @"Unsupported icon size");
+            return CGSizeMake(16, 16);
+    }
+}
+
++ (NSString *)stringForUnicodeCodepoint:(nullable NSString *)codepoint {
```

Comment:
Why don't you store the Unicode characters as-is in the JSON file? I.e. `"EA87" --> "\uEA87"`

Reply:
Primarily because I don't wish to change the json file, it's shared with React Native. Ideally we would just store a raw number instead for Objective-C

Reply:
Oh, ok, it makes sense then.